src/erlfmt.erl

```diff
@@ -408,27 +423,23 @@ read_nodes_string(FileName, String, Pragma) ->
 read_nodes(State, FileName, Pragma) ->
     read_nodes(State, FileName, Pragma, [], [], []).
 
-read_nodes({ok, Tokens, Comments, Cont}, FileName, Pragma, [], Warnings0, TextAcc) ->
+read_nodes({ok, Tokens, Comments, Cont}, FileName, PragmaFlag, [], Warnings0, TextAcc) ->
     {Node, Warnings} = parse_node(Tokens, Comments, FileName, Cont, Warnings0),
-    case {Pragma, contains_pragma_node(Node), Node} of
+    case {PragmaFlag, which_pragma_node(Node), Node} of
```

Comment:
Looks like we can remove the `PragmaFlag` from this case, it doesn't seem we're matching on it in any of the branches

Reply:
Great catch. I have updated the diff and removed it.